apps/backend/src/factory/StfcDownloadService.ts

```diff
@@ -0,0 +1,82 @@
+import querystring from 'querystring';
+
+import contentDisposition from 'content-disposition';
+import { Request, Response, NextFunction } from 'express';
+import { container } from 'tsyringe';
+
+import { Tokens } from '../config/Tokens';
+import { FileDataSource } from '../datasources/FileDataSource';
+import { Role } from '../models/Role';
+import {
+  MetaBase,
+  DownloadType,
+  PDFType,
+  XLSXType,
+  ZIPType,
+  DownloadService,
+  fetchDataAndStreamResponse,
+} from './DownloadService';
+import { ProposalPDFData } from './pdf/proposal';
+
+export class StfcDownloadService implements DownloadService {
+  async callFactoryService<TData, TMeta extends MetaBase>(
+    downloadType: DownloadType,
+    type: PDFType | XLSXType | ZIPType,
+    properties: { data: TData[]; meta: TMeta; userRole: Role },
+    req: Request,
+    res: Response,
+    next: NextFunction
+  ) {
+    let isPdfAvailable = false;
+    let proposalPdfData = null;
+    let fileName = '';
+
+    if (
+      properties?.data?.length === 1 &&
+      type === PDFType.PROPOSAL &&
+      downloadType === DownloadType.PDF
+    ) {
+      const data = properties?.data[0] as ProposalPDFData;
+
+      const fileDataSource = container.resolve<FileDataSource>(
+        Tokens.FileDataSource
+      );
+      proposalPdfData = await fileDataSource.getBlobdata(
+        `Migrated-${data.proposal.proposalId}.pdf`
+      );
+
+      if (proposalPdfData) {
+        isPdfAvailable = true;
+        fileName = `${data.proposal.proposalId}_${
+          data.principalInvestigator.lastname
+        }_${data.proposal.created.getUTCFullYear()}.pdf`;
+      }
+    }
```

Comment:
I am not seeing away we are handling multiple proposals and are we not going to allow zip for legacy proposals.

Reply:
Yes, currently its done only for single proposal pdf download. This was communicated with Emma.

Reply:
What happens if l send two proposals which are legacy and want zip

Reply:
It will follow our usual pdf download using pdf template one.  Focused more on users. User doesn't have the option for multi download I believe.

Reply:
This is not facing users only even the API can make a request to this end point.

Reply:
If we are not handling it we can just return a message for the  areas not implemented.

Reply:
ok sure. I have a doubt, since we are not introducing any new endpoint and for current scenarios, no scenario will be breaking, then is it suitable to return a message that its not implemented. For all other combinations, it will generate pdf normally using templates.

Reply:
Maybe we should open an issue for it if it not already open, because my understanding was that we can generate a proposal or get it as a legacy proposal from the database.